ui/restmodel/src/main/scala/pl/touk/nussknacker/restmodel/definition/package.scala

```diff
@@ -0,0 +1,79 @@
+import pl.touk.nussknacker.engine.definition.TypeInfos.MethodInfo
+import pl.touk.nussknacker.engine.graph.evaluatedparam
+import pl.touk.nussknacker.engine.graph.node.NodeData
+import pl.touk.nussknacker.restmodel.displayedgraph.displayablenode.EdgeType
+
+package object definition {
+
+  @JsonCodec(encodeOnly = true) case class UIProcessObjects(nodesToAdd: List[NodeGroup],
+                                                            processDefinition: UIProcessDefinition,
+                                                            nodesConfig: Map[String, SingleNodeConfig],
+                                                            additionalPropertiesConfig: Map[String, UiAdditionalPropertyConfig],
+                                                            edgesForNodes: List[NodeEdges],
+                                                            defaultAsyncInterpretation: Boolean)
+
+  @JsonCodec(encodeOnly = true) case class UIProcessDefinition(services: Map[String, UIObjectDefinition],
+                                 sourceFactories: Map[String, UIObjectDefinition],
+                                 sinkFactories: Map[String, UIObjectDefinition],
+                                 customStreamTransformers: Map[String, UIObjectDefinition],
+                                 signalsWithTransformers: Map[String, UIObjectDefinition],
+                                 exceptionHandlerFactory: UIObjectDefinition,
+                                 globalVariables: Map[String, UIObjectDefinition],
+                                 typesInformation: Set[UIClazzDefinition],
+                                 subprocessInputs: Map[String, UIObjectDefinition]) {
+    // skipping exceptionHandlerFactory
+    val allDefinitions: Map[String, UIObjectDefinition] = services ++ sourceFactories ++ sinkFactories ++
+      customStreamTransformers ++ signalsWithTransformers ++ globalVariables ++ subprocessInputs
+  }
+
+  @JsonCodec(encodeOnly = true) case class UIClazzDefinition(clazzName: TypingResult, methods: Map[String, MethodInfo])
+
+  @JsonCodec(encodeOnly = true) case class UIParameter(name: String,
+                                                       typ: TypingResult,
+                                                       editor: ParameterEditor,
+                                                       validators: List[ParameterValidator],
+                                                       additionalVariables: Map[String, TypingResult],
+                                                       branchParam: Boolean) {
+
+    def isOptional: Boolean = !validators.contains(MandatoryParameterValidator)
+
+  }
+
+  object UIParameter {
+
+    implicit def decoder(implicit typing: Decoder[TypingResult]): Decoder[UIParameter] = deriveDecoder[UIParameter]
+
+  }
+
+  @JsonCodec(encodeOnly = true) case class UIObjectDefinition(parameters: List[UIParameter],
+                                returnType: Option[TypingResult],
```

Comment:
Can you fix tabs so all variables will be on the same level? (I know that it was before)

Reply:
ok, done